src/xrGame/ui/UIMainIngameWnd.cpp

```diff
@@ -235,11 +235,10 @@ void CUIMainIngameWnd::Init()
         m_quick_slots_icons.push_back(new CUIStatic());
         m_quick_slots_icons.back()->SetAutoDelete(true);
         AttachChild(m_quick_slots_icons.back());
-        string32 path;
-        xr_sprintf(path, "quick_slot%d", i);
-        CUIXmlInit::InitStatic(uiXml, path, 0, m_quick_slots_icons.back());
-        xr_sprintf(path, "%s:counter", path);
-        UIHelper::CreateStatic(uiXml, path, m_quick_slots_icons.back());
+        std::string path = "quick_slot" + std::to_string(i);
+        CUIXmlInit::InitStatic(uiXml, path.c_str(), 0, m_quick_slots_icons.back());
+        path += ":counter";
+        UIHelper::CreateStatic(uiXml, path.c_str(), m_quick_slots_icons.back());
```

Comment:
Make it simplier, just use `xr_strcat` 😉 
```cpp
string32 path;
xr_sprintf(path, "quick_slot%d", i);
CUIXmlInit::InitStatic(uiXml, path, 0, m_quick_slots_icons.back());
xr_strcat(path, ":counter");
UIHelper::CreateStatic(uiXml, path, m_quick_slots_icons.back());
```

Reply:
xr_strcat is a C-style way, not C++. Also, using xr_strcat is less safe way. But you really need to correct the code - use xr_string instead of std::string.